cfme/tests/ansible/test_embedded_ansible_services.py

```diff
@@ -23,31 +26,47 @@
 SERVICE_CATALOG_VALUES = [
     ("default", None, "localhost"),
     ("blank", "", "localhost"),
-    ("unavailable_host", "unavailable_host", "unavailable_host")
+    ("unavailable_host", "unavailable_host", "unavailable_host"),
 ]
 
 
-@pytest.fixture(scope="module")
-def ansible_credential(appliance):
+CREDENTIALS = [
+    ("Amazon", "", "list_ec2_instances.yml"),
+    ("VMware", "vcenter_host", "gather_all_vms_from_vmware.yml"),
+    ("Red Hat Virtualization", "host", "connect_to_rhv.yml"),
+]
+
+
+@pytest.fixture
+def provider_credentials(appliance, provider, credential):
+    cred_type, hostname, playbook = credential
+    creds = provider.get_credentials_from_config(provider.data["credentials"])
+    credentials = {}
+    if cred_type == "Amazon":
+        credentials["access_key"] = creds.principal
+        credentials["secret_key"] = creds.secret
+    else:
+        credentials["username"] = creds.principal
+        credentials["password"] = creds.secret
+        credentials[hostname] = provider.hostname
+
     credential = appliance.collections.ansible_credentials.create(
-        fauxfactory.gen_alpha(),
-        "Machine",
-        username=fauxfactory.gen_alpha(),
-        password=fauxfactory.gen_alpha()
+        "{}_credential_{}".format(cred_type, fauxfactory.gen_alpha()),
+        cred_type,
+        **credentials
     )
     yield credential
 
```

Comment:
**Optional** can remove blank line here

Reply:
I have kept for the separation between `object yield` and `finalizer`